### ISSUE TITLE
For developers, have two sections: Detect and workaround

### ISSUE THREAD

**Issue by sandebert** [urbandroid-team/dont-kill-my-app#31]
The "Solution for developers" section is great, but even better would be if that section had two sub-sections:

- How to detect
- How to workaround

I imagine that there might be some cases where it's possible to detect, but not workaround. And in that case we could simply show the user some info, like ”This app is being choked by your phone, here's how to fix it blah blah blah”.

**Comment by ocram** [comment]
Right, on stock Android – and in general but with limited usefulness – you have `android.os.PowerManager#isDeviceIdleMode()` and `android.os.PowerManager#isIgnoringBatteryOptimizations(java.lang.String)`.

When it comes to detecting if you are affected by *any* (vendor-specific) optimizations, our go-to solution has always been setting some flags or storing logs during the execution of background tasks or on delivery of notifications, so that you can check for success when the app is in the foreground the next time.
